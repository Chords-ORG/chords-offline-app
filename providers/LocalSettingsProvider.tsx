import React from "react";
import { LightStyle, DarkStyle } from "../constants/Styles";
import useLocalConfiguration from "../hooks/useLocalConfiguration";
import { useColorScheme } from "react-native";
import { getItem } from "../functions/storage";

export type ChordType = "sharp" | "bemol";
export type Instrument = "guitar" | "piano";
export type CapoConfig = "auto" | "never";
export type LocalColorScheme = "system" | "light" | "dark";

export const CHORD_TYPE_KEY = "chordType";
export const INSTRUMENT_KEY = "instrument";
export const CAPO_CONFIG_KEY = "capoConfig";
export const LOCAL_COLOR_SCHEME_KEY = "localColorScheme";

interface LocalSettingsContextProps {
  chordType: ChordType;
  instrument: Instrument;
  capoConfig: CapoConfig;
  localColorScheme: LocalColorScheme;
  toggleLocalSettings: () => Promise<void>;
}

export const LocalSettingsContext =
  React.createContext<LocalSettingsContextProps>({
    chordType: "sharp",
    instrument: "guitar",
    capoConfig: "auto",
    localColorScheme: "system",
    toggleLocalSettings: async () => {},
  });

interface LocalSettingsProviderProps {
  children: React.ReactNode;
}

export const LocalSettingsProvider = ({
  children,
}: LocalSettingsProviderProps) => {
  const [chordType, setChordTypeState] = React.useState<ChordType>("sharp");
  const [instrument, setInstrumentState] = React.useState<Instrument>("guitar");
  const [capoConfig, setCapoConfigState] = React.useState<CapoConfig>("auto");
  const [localColorScheme, setLocalColorSchemeState] =
    React.useState<LocalColorScheme>("system");

  const getLocalConfiguration = async () => {
    const chordType = await getItem(CHORD_TYPE_KEY);
    const instrument = await getItem(INSTRUMENT_KEY);
    const capoConfig = await getItem(CAPO_CONFIG_KEY);
    const localColorScheme = await getItem(LOCAL_COLOR_SCHEME_KEY);
    return { chordType, instrument, capoConfig, localColorScheme };
  };

  const toggleLocalSettings = React.useCallback(async () => {
    const localConfiguration = await getLocalConfiguration();
    if (localConfiguration.chordType !== null) {
      setChordTypeState(localConfiguration.chordType as ChordType);
    }
    if (localConfiguration.instrument !== null) {
      setInstrumentState(localConfiguration.instrument as Instrument);
    }
    if (localConfiguration.capoConfig !== null) {
      setCapoConfigState(localConfiguration.capoConfig as CapoConfig);
    }
    if (localConfiguration.localColorScheme !== null) {
      setLocalColorSchemeState(
        localConfiguration.localColorScheme as LocalColorScheme
      );
    }
  }, [
    setChordTypeState,
    setInstrumentState,
    setCapoConfigState,
    setLocalColorSchemeState,
  ]);

  React.useEffect(() => {
    toggleLocalSettings();
  }, []);

  return (
    <LocalSettingsContext.Provider
      value={{
        chordType,
        instrument,
        capoConfig,
        localColorScheme,
        toggleLocalSettings,
      }}
    >
      {children}
    </LocalSettingsContext.Provider>
  );
};
