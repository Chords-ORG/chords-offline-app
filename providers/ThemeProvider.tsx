import React from "react";
import { LightStyle, DarkStyle } from "../constants/Styles";
import useLocalConfiguration from "../hooks/useLocalConfiguration";
import { useColorScheme } from "react-native";
import { getItem } from "../functions/storage";

interface ThemeContextProps {
  styleSheet: typeof LightStyle | typeof DarkStyle;
  theme: "light" | "dark";
  toggleTheme: () => void;
}

export const ThemeContext = React.createContext<ThemeContextProps>({
  styleSheet: LightStyle,
  theme: "light",
  toggleTheme: () => {},
});

interface ThemeProviderProps {
  children: React.ReactNode;
}
export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const [theme, setTheme] = React.useState<"light" | "dark">("light");
  const systemColorScheme = useColorScheme();
  const { localColorScheme } = useLocalConfiguration();

  const toggleTheme = async () => {
    if (localColorScheme === "system") {
      setTheme(systemColorScheme || "light");
    } else {
      setTheme(localColorScheme as "light" | "dark");
    }
  };

  React.useEffect(() => {
    toggleTheme();
  }, [systemColorScheme, localColorScheme]);

  return (
    <ThemeContext.Provider
      value={{
        styleSheet: theme === "light" ? LightStyle : DarkStyle,
        theme,
        toggleTheme,
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
};
