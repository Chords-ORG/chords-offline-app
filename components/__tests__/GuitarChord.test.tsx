import renderer from "react-test-renderer";
import React from "react";
import { ThemeContext } from "../../providers/ThemeProvider";
import {
  DarkColors,
  DarkStyle,
  LightColors,
  LightStyle,
} from "../../constants/Styles";
import GuitarChord from "../GuitarChord";

jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock")
);

describe("<GuitarChordImage/>", () => {
  [0, 4, 9].forEach((capo) => {
    it(`should render C on position 0 with capo ${capo} on light theme`, () => {
      const tree = renderer
        .create(
          <ThemeContext.Provider
            value={{
              styleSheet: LightStyle,
              colors: LightColors,
              theme: "light",
            }}
          >
            <GuitarChord chordName="C" capo={capo} />
          </ThemeContext.Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it(`should render C on position 0 with capo ${capo} on dark theme`, () => {
      const tree = renderer

        .create(
          <ThemeContext.Provider
            value={{
              styleSheet: DarkStyle,
              colors: DarkColors,
              theme: "dark",
            }}
          >
            <GuitarChord chordName="C" capo={capo} />
          </ThemeContext.Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
