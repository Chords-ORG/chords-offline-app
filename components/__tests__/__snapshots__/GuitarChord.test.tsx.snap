// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuitarChordImage/> should render C on position 0 with capo 0 on dark theme 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={2}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingRight": 10,
        "width": 100,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "roboto-bold",
            "fontSize": 12,
            "marginTop": 25,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
         
    </Text>
    <View>
      <Image
        source={1}
        style={
          {
            "height": 110,
            "width": 80,
          }
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 30,
            },
            {
              "right": 15,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 22,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 36,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 49,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 63,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 79,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 47,
            },
            {
              "right": 41,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 63,
            },
            {
              "right": 55,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 2,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 15,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 29,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 42,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 56,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 70,
            "width": 8,
          }
        }
      />
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`<GuitarChordImage/> should render C on position 0 with capo 0 on light theme 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingRight": 10,
        "width": 100,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "roboto-bold",
            "fontSize": 12,
            "marginTop": 25,
          },
          {
            "color": "#333333",
          },
        ]
      }
    >
         
    </Text>
    <View>
      <Image
        source={1}
        style={
          {
            "height": 110,
            "width": 80,
          }
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 30,
            },
            {
              "right": 15,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 22,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 36,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 49,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 63,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 79,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 47,
            },
            {
              "right": 41,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 63,
            },
            {
              "right": 55,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 2,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 15,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 29,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 42,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 56,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 70,
            "width": 8,
          }
        }
      />
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`<GuitarChordImage/> should render C on position 0 with capo 4 on dark theme 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={4}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingRight": 10,
        "width": 100,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "roboto-bold",
            "fontSize": 12,
            "marginTop": 25,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      5ª
    </Text>
    <View>
      <Image
        source={1}
        style={
          {
            "height": 110,
            "width": 80,
          }
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 30,
            },
            {
              "right": 15,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 22,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 36,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 49,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 63,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 79,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 47,
            },
            {
              "right": 41,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 63,
            },
            {
              "right": 55,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 2,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 15,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 29,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 42,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 56,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 70,
            "width": 8,
          }
        }
      />
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`<GuitarChordImage/> should render C on position 0 with capo 4 on light theme 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={3}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingRight": 10,
        "width": 100,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "roboto-bold",
            "fontSize": 12,
            "marginTop": 25,
          },
          {
            "color": "#333333",
          },
        ]
      }
    >
      5ª
    </Text>
    <View>
      <Image
        source={1}
        style={
          {
            "height": 110,
            "width": 80,
          }
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 30,
            },
            {
              "right": 15,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 22,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 36,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 49,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 63,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 79,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 47,
            },
            {
              "right": 41,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 63,
            },
            {
              "right": 55,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 2,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 15,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 29,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 42,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 56,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 70,
            "width": 8,
          }
        }
      />
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`<GuitarChordImage/> should render C on position 0 with capo 9 on dark theme 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={6}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingRight": 10,
        "width": 100,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "roboto-bold",
            "fontSize": 12,
            "marginTop": 25,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      10ª
    </Text>
    <View>
      <Image
        source={1}
        style={
          {
            "height": 110,
            "width": 80,
          }
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 30,
            },
            {
              "right": 15,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 22,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 36,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 49,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 63,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 79,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 47,
            },
            {
              "right": 41,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 63,
            },
            {
              "right": 55,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 2,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 15,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 29,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 42,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 56,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 70,
            "width": 8,
          }
        }
      />
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`<GuitarChordImage/> should render C on position 0 with capo 9 on light theme 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={5}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingRight": 10,
        "width": 100,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "roboto-bold",
            "fontSize": 12,
            "marginTop": 25,
          },
          {
            "color": "#333333",
          },
        ]
      }
    >
      10ª
    </Text>
    <View>
      <Image
        source={1}
        style={
          {
            "height": 110,
            "width": 80,
          }
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 30,
            },
            {
              "right": 15,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 22,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 36,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 49,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 63,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 7,
              "position": "absolute",
              "width": 79,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 47,
            },
            {
              "right": 41,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "top": 63,
            },
            {
              "right": 55,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          [
            {
              "height": 12,
              "position": "absolute",
              "width": 12,
            },
            {
              "height": 0,
              "width": 0,
            },
            {
              "height": 0,
              "width": 0,
            },
          ]
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 2,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 15,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 29,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 42,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 56,
            "width": 8,
          }
        }
      />
      <Image
        source={1}
        style={
          {
            "bottom": 0,
            "height": 8,
            "position": "absolute",
            "right": 70,
            "width": 8,
          }
        }
      />
    </View>
  </View>
</RNGestureHandlerButton>
`;
